music_assistant/providers/phishin/provider.py

```
@@ -0,0 +1,1015 @@
+        search_query: str,
+        media_types: list[MediaType],
+        limit: int = MAX_SEARCH_RESULTS,
+    ) -> SearchResults:
+        """Perform search on Phish.in."""
+        if len(search_query.strip()) < 3:
+            return SearchResults()
+
+        try:
+            endpoint = ENDPOINTS["search"].format(term=search_query)
+            search_data = await api_request(
+                self, endpoint, params={"audio_status": "complete_or_partial"}
+            )
+
+            # If venues are found and we want albums, fetch shows for those venues
+            if MediaType.ALBUM in media_types and search_data.get("venues"):
+                venue_shows: list[dict[str, Any]] = []
+                for venue in search_data.get("venues", []):
+                    venue_slug = venue["slug"]
+                    try:
+                        page = 1
+                        while len(venue_shows) < limit:
+                            shows_data = await api_request(
+                                self, "/shows", params={"venue_slug": venue_slug, "page": page}
+                            )
+
+                            shows_on_page = shows_data.get("shows", [])
+                            if not shows_on_page:
+                                break
+
+                            remaining_slots = limit - len(venue_shows)
+                            venue_shows.extend(shows_on_page[:remaining_slots])
+
+                            current_page = shows_data.get("current_page", 1)
+                            total_pages = shows_data.get("total_pages", 1)
+
+                            if current_page >= total_pages or len(venue_shows) >= limit:
+                                break
+
+                            page += 1
+
+                    except ProviderUnavailableError:
+                        self.logger.warning(
+                            "Failed to fetch shows for venue %s: API unavailable", venue_slug
+                        )
```

why not simply let the exception bubble up instead of swallowing it here silently ?
if you let the exception raise, the api handler will catch it and forward and error to the user that the action failed

---

hmm I see you kind of do this but then wrap everything in an outer exception, i think that can be deduplicated by just only have the outer exception handler which then wraps it into ProviderUnavailableError or MediaNotFoundError with a sensible error message.

---

Fixed